### DISCUSSION TITLE
Ветвление алгоритма в зависимости от дня недели

### DISCUSSION THREAD

**Discussion by amd64fox** [Chimildic/goofy#67] [Помощь с алгоритмом] [answered]
Запутался в простых вещах, хочу контролировать массив данных в зависимости от дня недели, например будет использоваться массив из Яндекс треков, пишу:

```javascript
    var t = 'понедельник'   // Например сегодня Понедельник

    if ( t = 'понедельник') {   // Если сегодня понедельник то хочу получить треки из 'плейлиста Дня' и 'Дежавю' 

    let tracks = Yandex.getTracks('yamusic-daily', '48633226');
    let tracks2= Yandex.getTracks('yamusic-dejavu', '35944899');
     
     Combiner.push(tracks, tracks2)
     
   } else {   // Если другое то хочу получить треки из 'плейлиста Дня' 
    
     let tracks = Yandex.getTracks('yamusic-daily', '48633226');
         
}

console.log(tracks.length)      // Ошибка  ReferenceError: tracks is not defined
```

Но за пределы условия массив не выходит, видно по ошибке, что массив `tracks` не определен.

Можно конечно в одной функции написать сразу 2 подфункции похожих  но с разными массивами и для каждой задать отдельное условие, но тогда размер кода увеличивается  сильно.

Есть ли способ как то научиться выпускать нужный массив в нужное время ? сорян если ахинею написал 😁

**Comment by Chimildic** [comment, marked as answer]
Поскольку в обоих случаях есть плейлист дня, можно сократить.
```js
let tracks = Yandex.getTracks('yamusic-daily', '48633226');
if (Selector.isDayOfWeekRu('понедельник')) {
    Combiner.push(tracks, Yandex.getTracks('yamusic-dejavu', '35944899'));
}
```
Но если это только для примера и подразумеваются совсем разные, то `tracks` объявить вначале. 
```js
let tracks = [];
if (Selector.isDayOfWeekRu('понедельник')){
    Combiner.push(tracks, Yandex.getTracks('yamusic-daily', '48633226'));
    Combiner.push(tracks, Yandex.getTracks('yamusic-dejavu', '35944899'));
} else {
    Combiner.push(tracks, Yandex.getTracks('yamusic-daily', '48633226'));
}
```
Также обращу внимание, что в условии используется двойное равенство `==`, а не одинарное `=`. Но в данном случае это не важно, поскольку используется функция `isDayOfWeekRu`.

**Reply by amd64fox** [reply]
Спасибо огромное,  помог и 1-й и 2-й вариант исполнения.

А есть какое-то логическое объяснение, почему нельзя оборачивать массив условием ?

**Reply by Chimildic** [reply]
`let` создает локальную переменную в области видимости родительских фигурных скобок `{}`, включая вложения.

В данном случае, переменная доступна только в рамках условия и не видна/не доступна после условия. Поэтому получаем ошибку о том, что переменная не определена, когда обращаемся к ней за пределами условия. 
```js
if (true){
  let x = 5;
  console.log(x); // нет ошибки
}
console.log(x); // ошибка
```
Объявляя переменную выше условия, ее областью видимости становится уже вся функция, то есть ее тело `{}`. Поэтому переменная доступна на всем протяжении функции включая вложения (условие). 
```js
let x;
if (true){
  x = 5;
  console.log(x); // нет ошибки
}
console.log(x); // нет ошибки
```

**Reply by amd64fox** [reply]
Наступила полная ясность, благодарю.
